Intersect (Core)/Extensions/EnumerableExtensions.cs

```diff
@@ -39,5 +44,77 @@ public static IEnumerable<T> SelectManyRecursive<T>(this IEnumerable<T> source,
 
             return result.Any() ? result.Concat(result.SelectManyRecursive(selector, cancellationToken)) : result;
         }
+
+        public static string GetDescription(this Enum value)
+        {
+            Type type = value.GetType();
+            string name = Enum.GetName(type, value);
+            if (name != null)
+            {
+                FieldInfo field = type.GetField(name);
+                if (field != null)
+                {
+                    DescriptionAttribute attr =
+                           Attribute.GetCustomAttribute(field,
+                             typeof(DescriptionAttribute)) as DescriptionAttribute;
+                    if (attr != null)
+                    {
+                        return attr.Description;
+                    }
+                }
+            }
+            return string.Empty;
+        }
+
+        public static string[] GetDescriptions(Type enumType, string except = null)
+        {
+            return Enum.GetValues(enumType)
+                .Cast<Enum>()
+                .Select(value => value.GetDescription())
+                .Where(value => string.IsNullOrEmpty(except) || value != except)
+                .ToArray();
+        }
+
+        public static GameObjectType GetRelatedTable(this Enum value)
+        {
+            Type type = value.GetType();
+            string name = Enum.GetName(type, value);
+            if (name != null)
+            {
+                FieldInfo field = type.GetField(name);
+                if (field != null)
+                {
+                    RelatedTable attr =
+                           Attribute.GetCustomAttribute(field,
+                             typeof(RelatedTable)) as RelatedTable;
+                    if (attr != null)
+                    {
+                        return attr.TableType;
+                    }
+                }
+            }
+            throw new ArgumentException($"{nameof(value)} did not have a valid RelatedTable attribute to pull from");
+        }
+
+        public static VariableType GetRelatedVariableType(this Enum value)
+        {
+            Type type = value.GetType();
+            string name = Enum.GetName(type, value);
+            if (name != null)
+            {
+                FieldInfo field = type.GetField(name);
+                if (field != null)
+                {
+                    RelatedVariableType attr =
+                           Attribute.GetCustomAttribute(field,
+                             typeof(RelatedVariableType)) as RelatedVariableType;
+                    if (attr != null)
+                    {
+                        return attr.VariableType;
+                    }
+                }
+            }
+            throw new ArgumentException($"{nameof(value)} did not have a valid RelatedVariableType attribute to pull from");
+        }
```

Comment:
These are extension methods for _enumerations_ not _enumerables_ and also are only ever relevant on `VariableMod` and `VariableType`. Move them to `VariableModExtensions` and `VariableTypeExtensions` and make the input parameter be strongly typed. Then you can use Enum.GetName<TEnum>() and typeof(TEnum) instead of these more generic types.

Rename `field` to `fieldInfo` too and  use `fieldInfo.GetCustomAttribute<RelatedVariableTypeAttribute>()` and `fieldInfo.GetCustomAttribute<RelatedTableAttribute>()`.

Reply:
Also applies to GetRelatedTable

Reply:
done